pkg/client/multi.go

```diff
@@ -40,6 +40,22 @@ func aggregate[T any, V ObjectPointer[T]](ctx context.Context, factory ProjectCl
 	return aggregateOptionalNaming[T, V](ctx, true, factory, cb)
 }
 
+// parseProject accepts an argument name that is either of the form resource or project::resource, and ClientFactory
+// If the name is not of project::resource form, it returns defaultProject, name, nil
+// otherwise, it parses off the project name and returns project, name without project::, nil
+func parseProject(ctx context.Context, name string, factory ProjectClientFactory) (pc Client, parsedArg string, err error) {
+	var parsedProject string
+	if possibleProject, possibleArg, found := strings.Cut(name, "::"); found {
+		parsedProject = possibleProject
+		parsedArg = possibleArg
+	} else {
+		parsedProject = factory.DefaultProject()
+		parsedArg = name
+	}
+	pc, err = factory.ForProject(ctx, parsedProject)
+	return pc, parsedArg, err
```

Comment:
No need to do an explicit return here since you are using named return variables (or whatever the term for that is, lol).